pom.xml

```diff
@@ -810,15 +810,6 @@
                         <reportNameSuffix>${surefire.reportNameSuffix}</reportNameSuffix>
                     </configuration>
                     <dependencies>
-                        <!--
-                             Force tests to execute with JUnit 4
-                             See http://maven.apache.org/surefire/maven-failsafe-plugin/examples/junit.html
-                         -->
-                        <dependency>
-                            <groupId>org.apache.maven.surefire</groupId>
-                            <artifactId>surefire-junit47</artifactId>
-                            <version>${version.surefire.plugin}</version>
-                        </dependency>
```

Comment:
> Before merging, we will need to check that the number of tests executed before this PR is consistent with the number of tests executed after.


Can you confirm everything was fine when you checked with the latest version of this PR? For posterity :)

Reply:
😄 yeah I was looking into it, actually. Seems like some tests got reported into a different group, so trying to figure out what happened (more like, what can we modify in the plugin config related to reporting, or should we even modify it)

Reply:
ok, so the tests are all executed. Cannot see why `ObjectProjectionBaseIT` and `CompositeProjectionBaseIT` get reported into root tests though...

Reply:
No idea, but while looking at those tests I saw that `SearchSetupHelper` was still using this `callOncePerClass` stuff that you removed from `MappingSetupHelper`. Though I'd mention it so that you can adjust this if necessary.

Reply:
@marko-bekhta Can we merge now? :)

Reply:
yes 😃 🎉